conversions/binary_to_decimal.c

```diff
@@ -1,24 +1,103 @@
+/** 
+ * @file
+ * @brief Converts a binary number to a decimal one.
+ * @details
+ * A binary number is an input, it is checked to be binary
+ * then, the number is converted to a decimal number.
+ * @author Kyler Smith
+ * @author [lazy-dude] (https://github.com/lazy-dude)
+ */
+
+// includes
+#include <assert.h> /// for assert
+#include <stdbool.h> /// for bool
+#include <stdint.h> /// for uintmax_t
+#include <stdio.h> /// for IO operations
+
 /**
- * Modified 07/12/2017, Kyler Smith
- *
+ * @brief is_binary checks whether num is a binary one
+ * @param num to be checked if it has binary representation
+ * @return boolean true if num is binary false if not
  */
+bool is_binary(uintmax_t num)
+{
+    unsigned remainder = 0;
 
-#include <stdio.h>
+    while (num > 0) {
+        remainder = num % 10;
+        if (remainder == 0 || remainder == 1) {
+            num /= 10;
+            continue;
+        } else
+            return false;
+    }
+    return true;
+}
 
-int main()
+/**
+ * @brief num_len finds length of an uintmax_t num
+ * @param num whose length to be computed
+ * @return i int length of num
+ */
+int num_len(uintmax_t num)
+{
+    int i;
+    for (i = 0; num > 0; i++) {
+        num /= 10;
+    }
+    return i;
+}
+
+/**
+ * @brief binary_decimal function does the actual job of conversion
+ * @param number binary to be converted
+ * @return decimal_number decimal representation of binary number
+ */
+uintmax_t binary_decimal(uintmax_t number)
 {
-    int remainder, number = 0, decimal_number = 0, temp = 1;
-    printf("\n Enter any binary number= ");
-    scanf("%d", &number);
+	unsigned remainder;
+	uintmax_t decimal_number = 0, temp = 1;
+
+    int length = num_len(UINTMAX_MAX) - 1;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Requires using the `inttypes.h` library. Check other parts of the code ([reference](https://en.cppreference.com/w/c/types/integer)). 🙂